README.md

```diff
@@ -1,24 +1,343 @@
+
+  _Body:_
+    ```json
+    {
+      "string_key": "names",
+      "label": "Names",
+      "custom_fields": {}
+
+    }
+    ```
+
+#### Update Vocabulary
+- **Request**
+
+  `PATCH /vocabularies/:string_key`
+
+  Only a vocabulary's `label` can be updated, `string_key` cannot be changed.
+
+- **Success Response**
+
+  _Status:_ 200
+
+  _Body:_
+  ```json
+  {
+     "string_key": "names",
+     "label": "Names",
+     "custom_fields": {}
+  }
+  ```
+
+#### Delete Vocabulary
+- **Request**
+
+  `DELETE /vocabularies/:string_key`
+
+- **Success Response**
+
+  _Status:_ 204
+
+  _Body:_ nil
+
+- **Error Response**
+  - If resource does not exist returns `404`
+  - If not successful return `500`
+
+
+#### Send back description/instruction
+- **Request**
+
+  `OPTIONS /vocabularies/`
+- **Success Response**
+  TODO
+
+### Vocabularies/Custom Fields
+#### Add Custom Field to Vocabulary
+- **Request**
+
+  `POST /vocabularies/:string_key/custom_fields`
+
+  _Required params:_ `field_key`, `label` and `data_type`
+
+- **Success Response**
+
+  _Status:_ 201
+
+  _Body:_ NOT SURE
+
+#### Update Custom Field
+- **Request**
+
+  `PATCH /vocabularies/:string_key/custom_fields/:field_key`
+
+  _Optional params:_ `label`
+
+  `field_key` and `data_type` cannot be changed after creation.
+
+- **Success Response**
+
+  _Status:_ 200
+
+  _Body:_ NOT SURE
+
+#### Delete Custom Field from Vocabulary
+- **Request**
+
+  `DELETE /vocabularies/:string_key/custom_fields/:field_key`
+
+  Deletes custom field from vocabulary object, fields within terms will not be deleted, they will just be ignored.
+
+- **Success Response**
+
+  _Status:_ 200
+
+  _Body:_ none
+
+
+### Vocabularies/Terms
+
+#### List/Search through Terms in Vocabulary
+- **Request**
+
+  `GET /vocabularies/:string_key/terms`
+
+  Examples:
+  ```
+  GET /vocabularies/:string_key/terms?page=1&per_page=20`
+  GET /vocabularies/:string_key/terms?q=smith&authority=naf&page=1&per_page=20`
+  GET /vocabularies/:string_key/terms?uri=https://creativecommons.org/publicdomain/zero/1.0/
+  GET /vocabularies/:string_key/terms?label=Smith # returns temp term b/c no uri was given
+  ```
+
+  Allowed query fields:
+    - `q`: query term labels using fuzzy matching
+    - `authority`: facet by authority
+    - `uri`: search by exact uri string
+    - `label`: search by exact label string
+
+  Other allowed params:
+    - `per_page`: number of terms per page, default: 20
+    - `page`: page to display
+
+- **Success Response**
+
+  Sorted primarily by match score (in the case of a q search), secondarily by label alphabetical sort
+
+  _Status:_ 200
+
+  _Body:_
+  ```json
+  {
+    "page": 1,
+    "per_page": 25,
+    "total records": 55,
+    "terms": [
+      {
+        "pref_label": 'Carla Galarza',
+        "uri":"https://id.library.columbia.edu/term/1111-2222-3333-...",
+        "authority": "local",
+        "custom_field_1": "something"
+      },
+      {
+        "pref_label": 'Eric O'Hanlon',
```

Comment:
```suggestion
        "pref_label": "Eric O'Hanlon",
```

Reply:
Github was also highlighting this value in red, so maybe it prefers double quotes.  Plus, double quotes around the name string are better for the apostrophe in my name, haha.